core/Http.php

```diff
@@ -742,7 +742,7 @@ public static function sendHttpRequestBy(
                 }
                 $response = '';
             } else {
-                $header = '';
+                $header = [];
```

Comment:
That's actually not correct. `explode` expects a string, so changing the default value will always create an error if no headers are given. The problem is more likely a few lines below.

```
                    $split = explode("\r\n\r\n", $response, 2);

                    if(count($split) == 2) {
                        [$header, $response] = $split;
                    } else {
                        $response = '';
                        $header = $split;
                    }
```

if the first explode has only one result, `$split` will be an array containing one element. That is than assigned to `$header` in the else. Guess it would be correct to use something like `$header = reset($split);` to ensure it will be the value of the first element and not the full array.

Reply:
Good spot @sgiehl, I'd got the types the wrong way around :roll_eyes: I was waiting on recreation data for this before testing which is now available - your suggested change works correctly when tested against the site where this issue was encountered. I've updated the PR :+1:

Reply:
sorry I set you wrong there!